### ISSUE TITLE
Suggestions are gone when suggestion box is closed

### ISSUE THREAD

**Issue by TheColorRed** [HvyIndustries/crane#163]
When typing `self::` Suggestions show up. If you close the suggestions in some way, such as click somewhere else or press <kbd>esc</kbd>. You can not get the suggestions back unless you type the last `:` again.

**Comment by TheColorRed** [comment]
I think we need to add a Completion provider (unless I am not seeing it).

```ts
const phpCompletion = languages.registerCompletionItemProvider(docFilter, new PhpCompletion, ':', '$', '>');
context.subscriptions.push(phpCompletion);
```

**Comment by nevadascout** [comment]
I believe there is code in the server to set up the completion provider (otherwise completion would not work at all?)

**Comment by TheColorRed** [comment]
Maybe it should be moved to the client?

**Comment by nevadascout** [comment]
I don't have a problem with that. It's only where it is now because the microsoft example code had it on the server side as we do

**Comment by nevadascout** [comment]
I have identified the cause of this and created a new issue to specifically fix this problem: #233
